[pro#363-3] General changes

gbp opened the pull request:
Fixes the following part of https://github.com/mysociety/alaveteli-professional/issues/363

> - [x] Probably replace OpenStruct with stub in XWithY decorator spec

Along with a few more issues I identified

1. Default styling of `a.button:visited` links appears as blue on blue
2. Returning an 404 when going to `/profile/subscriptions` seems the unnecessary confusing, so now redirects to `/plana/pro` if users doesn't have an active subscription

lizconlan commented on spec/models/alaveteli_pro/subscription_with_discount_spec.rb:
```diff
@@ -22,14 +22,14 @@ def mock_subscription(coupon)
 
     it 'applies a percentage discount correctly' do
-      coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+      coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
probably worth making Hound happy here as it's not going to be too painful

lizconlan commented on spec/models/alaveteli_pro/subscription_with_discount_spec.rb:
```diff
@@ -50,7 +50,7 @@ def mock_subscription(coupon)
 
       it 'returns false' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: false)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: false)
```
probably worth making Hound happy (as above)

lizconlan commented on spec/models/alaveteli_pro/subscription_with_discount_spec.rb:
```diff
@@ -61,7 +61,7 @@ def mock_subscription(coupon)
 
       it 'returns true' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
probably worth making Hound happy (as above)

lizconlan commented on spec/models/alaveteli_pro/subscription_with_discount_spec.rb:
```diff
@@ -76,7 +76,7 @@ def mock_subscription(coupon)
 
       it 'returns false' do
-        coupon = OpenStruct.new(id: "50_off", percent_off: 50, valid: true)
+        coupon = double(:coupon, id: "50_off", amount_off: nil, percent_off: 50, valid: true)
```
probably worth making Hound happy (as above)

lizconlan commented on spec/models/alaveteli_pro/subscription_with_discount_spec.rb:
```diff
@@ -87,7 +87,7 @@ def mock_subscription(coupon)
 
       it 'returns true' do
-        coupon = OpenStruct.new(id: "100_off", percent_off: 100, valid: true)
+        coupon = double(:coupon, id: "100_off", amount_off: nil, percent_off: 100, valid: true)
```
probably worth making Hound happy (as above) - sorry, these are basically my fault!

lizconlan commented on app/controllers/alaveteli_pro/base_controller.rb:
```diff
@@ -16,9 +16,9 @@ def pro_user_authenticated?(reason_params = nil)
+               pro_site_name: AlaveteliConfiguration.pro_site_name),
         email: _("Then you can access {{pro_site_name}}",
-                pro_site_name: AlaveteliConfiguration.pro_site_name)
+               pro_site_name: AlaveteliConfiguration.pro_site_name)
```
wrong way! should align under the double quote

lizconlan commented on app/controllers/alaveteli_pro/subscriptions_controller.rb:
```diff
@@ -124,4 +123,11 @@ def authenticate
+  def check_existing_subscriptions
+    # TODO: This doesn't take the plan in to account
+    unless @user.pro_account.try(:active?)
+      flash[:notice] = _('You aren\'t currently subscribed to any plans')
```
maybe the message should be `You don't currently have an active Pro subscription` as the terminology around plans isn't directly exposed to users - we can expand it to include whatever we call Batch when we get to it

gbp commented:
@lizconlan I've rebased onto develop as most of those hound warnings have already been dealt with on a different branch. Added a couple of fixups for the remaining issues.
